app/lib/server/server.ts

```diff
@@ -0,0 +1,116 @@
+import { JobID, JobRequest, TelegramAuth, Job } from '@/api'
+import { Delegation, SpaceDID } from '@storacha/ui-react'
+import { Client as StorachaClient, DID, Signer } from '@storacha/ui-react'
+import { TelegramClient } from '@/vendor/telegram'
+import { create as createObjectStorage } from '@/lib/store/object'
+import { create as createCipher } from '@/lib/aes-cbc-cipher'
+import { create as createRemoteStorage } from '@/lib/store/remote'
+import { create as createHandler } from './handler'
+import { create as createJobStorage } from '@/lib/store/jobs'
+import { AgentData } from '@storacha/access/agent'
+import { serviceConnection, getServerIdentity, receiptsEndpoint, getBotToken, telegramAPIID, telegramAPIHash } from './constants'
+import { validate as validateInitData, parse as parseInitData } from '@telegram-apps/init-data-node';
+import { StringSession } from '@/vendor/telegram/sessions'
+import { TelegramClientParams } from '@/vendor/telegram/client/telegramBaseClient'
+import { Name } from '@storacha/ucn'
+
+const defaultClientParams: TelegramClientParams = { connectionRetries: 5 }
+
+export interface Context {
+    queueFn: (jr: JobRequest) => Promise<void>
+}
+
+export const create = async (ctx: Context) => {
+  return new JobServer(ctx)
+}
+
+class JobServer {
+
+  #queueFn
+
+  constructor ({  queueFn }: Context) {
+
+    this.#queueFn = queueFn
+  }
+
+  async queueJob(request: JobRequest) {
+    const handler = await this.#initializeHandler(request)
+    return await handler.queueJob(request)
+  }
+
+  async handleJob(request: JobRequest) {
+    const handler = await this.#initializeHandler(request)
+    return await handler.handleJob(request)
+  }
+
+  async #initializeHandler(request: JobRequest) {
+    const storacha = this.#initializeStoracha(request.spaceDID, request.spaceDelegation)
+    const telegram = await this.#initializeTelegram(request.telegramAuth)
+    const cipher = createCipher(request.encryptionPassword)
+    const name = Name.from(getServerIdentity(), [request.nameDelegation])
+    const remoteStore = createRemoteStorage(storacha)
+    const store = createObjectStorage<Record<JobID, Job>>({ remoteStore, name, cipher })
+    const jobs = await createJobStorage({ store })
+    return createHandler({
+      storacha, telegram, cipher, jobs, queueFn: this.#queueFn
+    })
+  }
+
+  async #initializeTelegram(telegramAuth: TelegramAuth) {
+    validateInitData(telegramAuth.initData, getBotToken())
+    const initData = parseInitData(telegramAuth.initData)
+    const session = new StringSession(telegramAuth.session) 
+    const client = new TelegramClient(session, telegramAPIID, telegramAPIHash, defaultClientParams)
+    if (!(await client.connect())) {
+      throw new Error("failed to connect to telegram")
+    }
+    if (!(await client.checkAuthorization())) {
+      throw new Error("client authorization failed")
+    }
+    const user = await client.getMe()
+    if (user.id !== BigInt(initData.user?.id || 0)) {
```

Comment:
Maybe only doing this check in `prod`, since things can be different in `dev`. For some reason, `initUser` is my test telegram account and `user` is my actual real one.